help: Document org setting to allow viewing only message moves.

laurynmm opened the pull request:
Documents organization setting that configures what message edit history can be viewed: content edits and topic/channel moves, only topic/channel moves, or disabling viewing edit history completely.

Fixes #33684.

---

**Notes/Questions**:
- There might be a better title for the article instead of "Configure viewing message edit history". It's a bit long in the left sidebar.
  - I didn't change the name of the file in this first round as that will likely depend on what this new article title is.
- Do we want the "?" in the organization settings panel for this setting? It's currently the only setting on that panel with any punctuation.

---

**Screenshots and screen captures:**

*There's no screenshot below if the changes were only to the related articles section of the help center article.*

Left sidebar changes:
![Screenshot from 2025-03-06 21-08-30](https://github.com/user-attachments/assets/8c04d921-58a2-4435-9bb4-0ff63e4d01ab)

| Location | Screenshot |
| --- | --- |
| [Configure viewing message edit history](https://zulip.com/help/disable-message-edit-history) | ![Screenshot from 2025-03-06 21-06-45](https://github.com/user-attachments/assets/56a2dfcb-5063-439a-afa9-8d4bc1094647) |
| [View message edit history](https://zulip.com/help/view-a-messages-edit-history) | ![Screenshot from 2025-03-06 21-06-51](https://github.com/user-attachments/assets/d7b663ba-4e4c-493f-abfc-7e88a4dd124c) |
| [Edit a message](https://zulip.com/help/edit-a-message) | ![Screenshot from 2025-03-06 21-08-02](https://github.com/user-attachments/assets/cbfd3c68-ef63-4605-a903-92a708605327) |
| [Keyboard shortcuts - selected message section](https://zulip.com/help/keyboard-shortcuts#for-a-selected-message-outlined-in-blue) | ![Screenshot from 2025-03-06 21-05-49](https://github.com/user-attachments/assets/7ca93798-f13a-462a-941c-2843e7f0ff2c) |

---

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

timabbott commented on help/view-a-messages-edit-history.md:
```diff
@@ -1,15 +1,16 @@
+channel](/help/move-content-to-another-channel) or
+[topic](/help/move-content-to-another-topic) are labeled **MOVED**.
+In organizations where viewing message edit history is
+disabled completely, moved messages are simply marked as **EDITED**.
```
I think we get to remove this last sentence when we do #33872.

alya commented:
All right, I pushed my proposed changes here.

Note that we now write "views", not "VIEWS", for the all-caps font style. So I think "moved" and "edited" should get the same treatment.


![Screenshot 2025-03-07 at 10 49 18](https://github.com/user-attachments/assets/d676aa49-eb27-4b32-8464-8b409d3118d0)
![Screenshot 2025-03-07 at 10 49 25](https://github.com/user-attachments/assets/3eb1cc7f-3096-4fa8-b02f-294fdc0a4bad)

laurynmm commented:
**Updates**:
- Added a prep commit to update the open graph test in `zerver/tests/test_middleware.py` to use a different help center article.
- Changed the markdown file name and added a URL redirect for the new help center article title: "Restrict message edit history access".
- Added a final commit that takes into account the changes in #33872. If we want to merge the bulk of the changes here before that's completed, then that final commit can be dropped and done in a separate pull request.

alya commented:
Cool! @timabbott Marking for integration review, but note that the last commit strictly speaking shouldn't be merged yet.

timabbott commented:
Merged, thanks @laurynmm! I think we're quite likely to complete #33872 for this release, and it seems pretty likely in expectation to be close to 0 actual users who experience confusion from merging that last commit a bit early giving Cloud deployment lag, so I didn't think it worth the tracking cost of not merging.
